File: machine_precision_1.py
import math

# True value of e
true_value_e = math.e

# Approximated value with 4 digits for the mantissa and 2 digits for the exponent
approximated_value = 2.7183 * 10**0  # Using 2.7183 as the mantissa and 10^0 as the exponent

# Absolute Error
absolute_error = abs(true_value_e - approximated_value)

# Relative Error
relative_error = absolute_error / true_value_e

# Display the results
print(f"True Value of e: {true_value_e}")
print(f"Approximated Value: {approximated_value}")
print(f"Absolute Error: {absolute_error}")
print(f"Relative Error: {relative_error}")

File: machine_precision_1.c
#include <stdio.h>
#include <float.h>
#include <math.h>

int main() {
    // Machine precision (epsilon) for the C float data type
    float machine_precision = FLT_EPSILON;

    // Storing π in a float variable
    float pi_stored = (float)M_PI;

    // True value of π
    double pi_true = M_PI;

    // Relative error in storing π
    float relative_error = fabs((pi_stored - pi_true) / pi_true);

    // Display the results
    printf("Machine Precision (float): %e\n", machine_precision);
    printf("Storing π in float: %.10f\n", pi_stored);
    printf("Relative Error in Storing π: %e\n", relative_error);

    return 0;
}
